cmd/juju/application/refresh.go

```diff
@@ -361,6 +366,16 @@ func (c *refreshCommand) Run(ctx *cmd.Context) error {
 	if err != nil {
 		return errors.Trace(err)
 	}
+	newOrigin := oldOrigin
+	if c.Base != "" {
+		if newBase, err := corebase.ParseBaseFromString(c.Base); err != nil {
+			return errors.Trace(err)
+		} else {
+			newOrigin.Base.OS = newBase.OS
+			newOrigin.Base.Channel = newBase.Channel
```

Comment:
This opens up refreshing machine charms to revisions which do not support the operating system  installed on the machine. Please document the implication for machine charms in the help, be careful what you wish for.